### DISCUSSION TITLE
Color Background of each Row,  Help!

### DISCUSSION THREAD

**Discussion by oz-hjx210** [filamentphp/filament#6044] [Help] [answered]
SORRY,How can I implement alternate row coloring?

   When I wrote the code in this way, it worked.

   protected function getTableRecordClassesUsing(): ?Closure
    {

        return function (Prod $record) {    
           if($record->ishome=='1'){
                return  'bg-primary-500';
            }else{
                return  'bg-gray-500';
            } 
       }
}

 But when I wrote it like this, it didn't work.
   protected function getTableRecordClassesUsing(): ?Closure
    {

        return function (Prod $record) {    
  
             self::$i++;
           if(self::$i%2==1){
                return  'bg-primary-500';
            }else{
               return  'bg-gray-500';
            }
        }
   }
    
I want to alternate row colors to differentiate every other row.
How should I use it correctly?  Tahnks!

**Comment by danharrin** [comment]
Make sure you're using a custom admin panel theme, as those background color classes might not exist in Tailwind yet.

**Reply by oz-hjx210** [reply]
those background color classes  is  exist  
 is any demo code teach me How should I use it correctly?

**Reply by oz-hjx210** [reply]
![image](https://user-images.githubusercontent.com/120376499/228425416-12ee4611-6bc5-46f2-a748-18478a8e1e6b.png)

**Comment by oz-hjx210** [comment, marked as answer]
public   static  $i=1;
    public   static  $nowid='';
   
 protected function getTableRecordClassesUsing(): ?Closure
    {
        return function (Prod $record) {
                        if($record->id!=self::$nowid){
                            self::$nowid=$record->id;
                            self::$i++;
                        }
                        if(self::$i%2==1){
                         return  'bg-primary-500';;
                       }else{
                         return  'bg-gray-500';;
                       }
         
       };
   }
I  find  the reason  , ,because the code excute twice,so i  add an IF ,like this   it works
